presto-main/src/main/java/com/facebook/presto/execution/SqlTaskManager.java

```diff
@@ -568,4 +580,42 @@ public QueryContext getQueryContext(QueryId queryId)
     {
         return queryContexts.getUnchecked(queryId);
     }
+
+    @Override
+    public BooleanIntImmutablePair shouldBackPressure(TaskUpdateRequest taskUpdateRequest)
```

Comment:
What if you create a `TaskBackpressureCircuitBreaker`, which is essentially this method, and where you wire in the configs necessary for better encapsulation?